server/controllers/finance/reports/financial.patient.js

```diff
@@ -41,13 +43,29 @@ function build(req, res, next) {
   }
 
   const data = {};
+  data.includeStockDistributed = parseInt(options.include_stock_distributed, 10);
 
   return Patients.lookupPatient(req.params.uuid)
     .then(patient => {
       _.extend(data, { patient });
-      return Debtors.getFinancialActivity(patient.debtor_uuid, true);
+      const dbPromises = [
+        Debtors.getFinancialActivity(patient.debtor_uuid, true),
+      ];
+
+      if (data.includeStockDistributed) {
+        dbPromises.push(Patients.stockMovementByPatient(req.params.uuid));
+      }
+
+      return q.all(dbPromises);
     })
-    .then(({ transactions, aggregates }) => {
+    .then(results => {
+      const { transactions } = results[0];
+      const { aggregates } = results[0];
```

Comment:
Just FYI - you can also combine these: 
```js
const { transactions, aggregates } = results[0];
```
or 
```js
const [{transactions, aggregate}] = results;
```

This shortcut is really confusing though for people who aren't used to ECMAScript 2019.  I don't really care which way it is done, but just wanted you to be aware of it :)